src/__tests__/non_regression.test.ts

```diff
@@ -0,0 +1,72 @@
+import { createReadStream } from 'node:fs';
+import { writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { Readable } from 'node:stream';
+
+import { expect, test } from 'vitest';
+
+import { FileCollection } from '../FileCollection.js';
+import { getZipReader } from '../zip/get_zip_reader.js';
+
+/**
+ * Run once on v5.2.2 to generate a legacy archive, pre-fix path encoding for filesystems.
+ * This test must not be edited or removed.
+ */
+test.fails('Generate a ium archive with problematic characters', async () => {
```

Comment:
This test document how `v5.2.2.ium.zip` had been generated and reminds the relativePath of sources on v5.2.2 and before.

@targos